File: string_array/Remove_Duplicates_from_Sorted_Array_II.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int k = 1;
        int track = nums[0];
        boolean flag = true;

        for (int i = 1; i < nums.length; ++i) {
            if (flag && nums[i] == track) {
                nums[k++] = nums[i];
                flag = false;
            }
            else if (nums[i] != track) {
                track = nums[i];
                nums[k++] = nums[i];
                flag = true;
            }
        }
        return k;
    }
}

File: string_array/Remove_Duplicates_from_Sorted_Array_II.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        k = 1
        track = nums[0]
        flag = True

        for i in range(1, len(nums)):
            if flag and nums[i] == track:
                nums[k] = nums[i]
                flag = False
                k += 1
            elif nums[i] != track:
                track = nums[i]
                nums[k] = nums[i]
                flag = True
                k += 1
        return k